plugins/out_kinesis_streams/kinesis_api.c

```diff
@@ -97,6 +97,47 @@ static int init_put_payload(struct flb_kinesis *ctx, struct flush *buf,
     return -1;
 }
 
+/*
+ * Adjust the time_key output format according to time_key_format.
+ */
+static void get_time_format_string(char *str, size_t maxsize,
+                      const char *time_key_format, const struct tm *timeptr,
+                      size_t len, struct flb_time *tms){
+    int millisecond_length = ((int)((ceil(log10(abs((uint64_t) tms->tm.tv_nsec / 1000000)))+1)*sizeof(char)));
+    int nanosecond_length = ((int)((ceil(log10(abs((uint64_t) tms->tm.tv_nsec)))+1)*sizeof(char)));
+    char millisecond_string[millisecond_length];
+    char nanosecond_string[nanosecond_length];
+    const char *mem_for_time_key_format;
+    strcpy(mem_for_time_key_format, time_key_format);
+    snprintf(millisecond_string, millisecond_length, "%" PRIu64, (uint64_t) tms->tm.tv_nsec / 1000000);
+    snprintf(nanosecond_string, nanosecond_length, "%" PRIu64, (uint64_t) tms->tm.tv_nsec);
+    char *milli_string_old = "%F";
+    char *nano_string_old = "%L";
+    char bstr[strlen(time_key_format)];
+    memset(bstr,0,sizeof(bstr));
+    for (int i = 0; i < strlen(time_key_format); i++) {
+        if (!strncmp(time_key_format + i, milli_string_old, strlen(milli_string_old))) {
+            strcat(bstr, millisecond_string);
+            i += strlen(milli_string_old) - 1;
+        } else {
+            strncat(bstr,time_key_format + i,1);
+        }
+    }
+    strcpy(time_key_format, bstr);
+    memset(bstr,0,sizeof(bstr));
+    for (int i = 0; i < strlen(time_key_format); i++) {
+        if (!strncmp(time_key_format + i, nano_string_old, strlen(nano_string_old))) {
+            strcat(bstr, nanosecond_string);
+            i += strlen(nano_string_old) - 1;
+        } else {
+            strncat(bstr,time_key_format + i,1);
+        }
+    }
+    strcpy(time_key_format, bstr);
+    strftime(str, maxsize, time_key_format, timeptr);
+    strcpy(time_key_format, mem_for_time_key_format);
```

Comment:
This is kinda a weird pattern. You're using some input variable as a temporary variable and then restoring it later. If you want to use a temporary variable, you shouldn't use an argument variable

Reply:
So if the data (the final time string) is "leaving" this function via the `time_key_format` variable... that is potentially fine... but the variable name should make it more clear. FB would typically use a name like `out_buf`.